### ISSUE TITLE
Training code hangs after val cache creation

### ISSUE THREAD

**Issue by PuneethBC** [ultralytics/ultralytics#22140]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

version 8.3.202
yolov11 nano training with pretrained model initialization.

Training code hangs after this line
val: New cache created: <>\datasets\labels\val.cache

I am running with albumentations enabled.

### Environment

Ultralytics 8.3.202  Python-3.13.0 torch-2.8.0+cu129 CUDA:0 (NVIDIA RTX A4500 Laptop GPU, 16384MiB)
Setup complete  (24 CPUs, 31.6 GB RAM, 859.6/952.3 GB disk)

OS                     Windows-10-10.0.19045-SP0
Environment            Windows
Python                 3.13.0
Install                pip
Path                   <my_venv>\Lib\site-packages\ultralytics
RAM                    31.61 GB
Disk                   859.6/952.3 GB
CPU                    12th Gen Intel Core(TM) i7-12850HX
CPU count              24
GPU                    NVIDIA RTX A4500 Laptop GPU, 16384MiB
GPU count              1
CUDA                   12.9

numpy                   2.2.6>=1.23.0
matplotlib              3.10.6>=3.3.0
opencv-python           4.12.0.88>=4.6.0
pillow                  11.3.0>=7.1.2
pyyaml                  6.0.2>=5.3.1
requests                2.32.5>=2.23.0
scipy                   1.16.1>=1.4.1
torch                   2.8.0+cu129>=1.8.0
torch                   2.8.0+cu129!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision             0.23.0+cu129>=0.9.0
psutil                  7.0.0
polars                  1.33.1
ultralytics-thop        2.0.17>=2.0.0
{'OS': 'Windows-10-10.0.19045-SP0', 'Environment': 'Windows', 'Python': '3.13.0', 'Install': 'pip', 'Path': '<my_venv>\\Lib\\site-packages\\ultralytics', 'RAM': '31.61 GB', 'Disk': '859.6/952.3 GB', 'CPU': '12th Gen Intel Core(TM) i7-12850HX', 'CPU count': 24, 'GPU': 'NVIDIA RTX A4500 Laptop GPU, 16384MiB', 'GPU count': 1, 'CUDA': '12.9', 'Package Info': {'numpy': '✅ 2.2.6>=1.23.0', 'matplotlib': '✅ 3.10.6>=3.3.0', 'opencv-python': '✅ 4.12.0.88>=4.6.0', 'pillow': '✅ 11.3.0>=7.1.2', 'pyyaml': '✅ 6.0.2>=5.3.1', 'requests': '✅ 2.32.5>=2.23.0', 'scipy': '✅ 1.16.1>=1.4.1', 'torch': '✅ 2.8.0+cu129!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.23.0+cu129>=0.9.0', 'psutil': '✅ 7.0.0', 'polars': '✅ 1.33.1', 'ultralytics-thop': '✅ 2.0.17>=2.0.0'}}

### Minimal Reproducible Example

Iteratively train the model by reloading the previous model

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @PuneethBC, thanks for opening this issue and for using Ultralytics 🚀 This is an automated response to help get things moving quickly — an Ultralytics engineer will follow up soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Since your training appears to hang after val cache creation with YOLO11 nano and albumentations enabled, please include:
-Exact train command or Python snippet used
-Full console log with any warnings before the hang (run with `verbose=True` if using Python or `ultralytics.settings["runs_dir"]` path to logs)
-A tiny sample of your dataset and the dataset YAML (enough to reproduce)
-Confirmation whether the issue reproduces with albumentations disabled
-Whether it occurs from a fresh run vs. when reloading from a previous model, and the exact weights path
-The created `val.cache` file size and whether deleting it changes behavior

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for the detailed environment info 🙏 Once you share an MRE and the requested details, we’ll help investigate further.

**Comment by Y-T-G** [comment]
You need to provide a minimum reproducible example and the code you used.

**Comment by Y-T-G** [comment]
Also try downgrading to Python 3.12. Python 3.13 is not recommended
